### ISSUE TITLE
Clusters convolutions with different kernels

### ISSUE THREAD

**Issue by stared** [waleedka/hiddenlayer#23]
I use the newest version  bd0d36a475279353a3a093a3954752f04288283a 

```{python}
import torch
from torch import nn
import hiddenlayer as hl

model = nn.Sequential(
          nn.Conv2d(8, 8, 3, padding=1),
          nn.ReLU(),
          nn.Conv2d(8, 8, 1),
          nn.ReLU(),
          nn.MaxPool2d(2, 2))

hl.build_graph(model, torch.zeros([1, 8, 32, 32]))
```

![screenshot 2019-01-19 07 54 29](https://user-images.githubusercontent.com/1001610/51423401-d6535600-1bbf-11e9-80b9-35a181d52529.png)

When there are different architectures

```{python}
model = nn.Sequential(
          nn.Conv2d(8, 8, 3, padding=1),
          nn.ReLU(),
          nn.Conv2d(8, 8, 1),
          nn.MaxPool2d(2, 2))
```

![screenshot 2019-01-19 07 55 15](https://user-images.githubusercontent.com/1001610/51423400-d6535600-1bbf-11e9-8875-87f8718577d8.png)

### Side notes

* It clusters ReLU with operations (which I like a lot, otherwise there is too much of visual noise), but not other activation functions (sigmoid, tanh, etc); is there some rationale for that? (My preference would be to cluster all reasonable activation functions).
* In general, I am very enthusiastic about neural network visualizations, see my overview: [Simple diagrams of convoluted neural networks](https://medium.com/inbrowserai/simple-diagrams-of-convoluted-neural-networks-39c097d2925b); I did mine, but for a limited case od purely sequential networks in Keras: https://github.com/stared/keras-sequential-ascii,

**Comment by stared** [comment]
Thanks, @waleedka !

I see it uses Graphviz (fortunately, now it is easy to install it it with `conda`; previously it was painful), open ot developing a more interactive version. (And ideally, a version capable of displaying this hierarchical structure, as "Inception block.")

I would be happy to talk more, since there is an overlap in interests and approaches. Vide my https://github.com/stared/livelossplot package (did you take any inspiration from it?) and to a lesser extend, variants of confusion matrix at https://github.com/stared/keras-interactively-piterpy2018/blob/master/talk.ipynb.

**Comment by plet1** [comment]
I have problem at the last line: **_hl.build_graph(model, torch.zeros(????????????????))_** 
If we have this model
"
model = nn.Sequential(
          nn.Conv2d(8, 8, 3, padding=1),
          nn.ReLU(),
          nn.Conv2d(8, 8, 1),
          nn.ReLU(),
          nn.MaxPool2d(2, 2))

model_1= nn.Sequential(
          nn.Conv2d(8, 8, 3, padding=1),
          nn.ReLU(),
          nn.Conv2d(8, 8, 1),
          nn.ReLU(),
          nn.MaxPool2d(2, 2))  
hl.build_graph(model, torch.zeros(_**????????????????**_))  "

**_Should we write  this line "hl.build_graph(model, torch.zeros(??????)) ""  ?_** 
thank you. 
p/s: I also tried : hl.build_graph(model, torch.zeros([1, 8, 32, 32], [1, 8, 32, 32])) . But not right.

**Comment by stared** [comment]
@waleedka Thank you! So, it is awesome to learn that we can to the same ideas... and make them work.

Do you think it would be a good idea to join forces? (One for tracking the training process, another for graph viz.)

**Comment by waleedka** [comment]
@stared Yes, happy to talk about it. I'll DM you.
